packages/core/src/domain/IArtifact.ts

```diff
@@ -12,3 +12,6 @@ export interface IArtifact {
   kind?: string; // TODO delete
   customKind?: boolean; // TODO delete
 }
+
+export const ARTIFACT_TYPE_EMBEDDED = 'embedded/base64';
+export const ARTIFACT_TYPE_REMOTE = 'remote/base64';
```

Comment:
Duplicates `ArtifactTypes.ts`?

Reply:
`ArtifactTypes.ts` are used for pattern matching which we don't do here. I mean we could, but it seems unnecessary to pattern match when we know the full type. Honestly, I dont even know why we pattern match at all lol. There could be some reasoning for future proofing, but I've never seen it used, and this type of future proofing doesn't seem too useful, tbh, but I could be wrong. I don't know deck all that well, but the pattern matching is questionable, considering the backend rarely does it. I think it's only in one spot in clouddriver lol, but everything else checks directly against the type.